### PR TITLE
refactor: update changes according to #708

### PR DESCRIPTION
**Pull request by deimosfr**

### CODE REVIEW

File: src/dns_provider/cloudflare.rs [hunk 1]
```diff
@@ -2,7 +2,8 @@ use std::net::Ipv4Addr;
 use tera::Context as TeraContext;
 
 use crate::dns_provider::errors::DnsProviderError;
-use crate::dns_provider::{DnsProvider, DnsProviderConfiguration, Kind};
+use crate::dns_provider::io::Kind;
```
Comment (benjaminch) [inline src/dns_provider/cloudflare.rs:5]: should be `use crate::dns_provider::Kind;`

File: src/dns_provider/io.rs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
+pub enum Kind {
+    Cloudflare,
+    QoveryDns,
+}
```
Comment (benjaminch) [inline src/dns_provider/io.rs:8]: Add From impl to convert this one to domain model.

```
use crate::dns_provider;
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize, Clone, Debug)]
#[serde(rename_all = "SCREAMING_SNAKE_CASE")]
pub enum Kind {
    Cloudflare,
    QoveryDns,
}

impl From<dns_provider::Kind> for Kind {
    fn from(kind: dns_provider::Kind) -> Self {
        match kind {
            dns_provider::Kind::Cloudflare => Kind::Cloudflare,
            dns_provider::Kind::QoveryDns => Kind::QoveryDns,
        }
    }
}
```

File: src/dns_provider/mod.rs [hunk 3]
```diff
@@ -3,22 +3,16 @@ use std::net::Ipv4Addr;
...
 use tera::Context as TeraContext;
 
 use crate::io_models::{Context, Domain};
 
 pub mod cloudflare;
 pub mod errors;
+mod io;
 pub mod qoverydns;
 
-#[derive(Serialize, Deserialize, Clone, Debug)]
```
[the remark this answers is not in the document]
Comment (benjaminch) [reply src/dns_provider/mod.rs:15]: It's your domain model

File: src/dns_provider/io.rs (hunk 2 above)
Comment (benjaminch) [reply src/dns_provider/io.rs:8]: It's your IO model

File: src/dns_provider/qoverydns.rs [hunk 4]
```diff
@@ -2,7 +2,8 @@ use std::net::Ipv4Addr;
 use tera::Context as TeraContext;
 
 use crate::dns_provider::errors::DnsProviderError;
-use crate::dns_provider::{DnsProvider, DnsProviderConfiguration, Kind};
+use crate::dns_provider::io::Kind;
```
Comment (benjaminch) [inline src/dns_provider/qoverydns.rs:5]: should be `use crate::dns_provider::Kind;`

File: src/dns_provider/mod.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (deimosfr) [reply src/dns_provider/mod.rs:15]: I don't find the facepalm icon. Let me update it sorry

**Pull request closed without merging**